tests/scripts/payment/mark_payments_as_sent_test.py

```diff
@@ -0,0 +1,52 @@
+from pcapi.models.payment_status import PaymentStatus
+from pcapi.models.payment_status import TransactionStatus
+from pcapi.scripts.payment.mark_payments_as_sent import mark_payments_as_sent
+
+
+class MarkPaymentsAsSentTest:
+    def test_mark_payments_as_sent(self, app, db_session):
+        # Given
+        payment1 = payment_factories.PaymentFactory(id=100, statuses=[], transactionLabel="transaction_label")
+        payment2 = payment_factories.PaymentFactory(id=101, statuses=[], transactionLabel="transaction_label")
+        payment3 = payment_factories.PaymentFactory(id=102, statuses=[], transactionLabel="transaction_label")
+        payment4 = payment_factories.PaymentFactory(id=103, statuses=[], transactionLabel="transaction_label")
+        no_status_under_review_payment = payment_factories.PaymentFactory(
+            id=104, statuses=[], transactionLabel="transaction_label"
+        )
+        wrong_transaction_label_payment = payment_factories.PaymentFactory(
+            id=105, statuses=[], transactionLabel="wrong_transaction_label"
+        )
+        payments_to_be_marked_as_sent = [payment1, payment2, payment3, payment4]
+        payment_factories.PaymentStatusFactory(payment=payment1, status=TransactionStatus.UNDER_REVIEW)
+        payment_factories.PaymentStatusFactory(payment=payment1, status=TransactionStatus.PENDING)
+        payment_factories.PaymentStatusFactory(payment=payment2, status=TransactionStatus.UNDER_REVIEW)
+        payment_factories.PaymentStatusFactory(payment=payment3, status=TransactionStatus.UNDER_REVIEW)
+        payment_factories.PaymentStatusFactory(payment=payment4, status=TransactionStatus.UNDER_REVIEW)
+        payment_factories.PaymentStatusFactory(payment=no_status_under_review_payment, status=TransactionStatus.PENDING)
+
+        # When
+        mark_payments_as_sent("transaction_label", 2)
+
+        # Then
+        assert PaymentStatus.query.count() == 6 + 4
+        payments_statuses_sent = PaymentStatus.query.filter(PaymentStatus.status == TransactionStatus.SENT).all()
+        assert len(payments_statuses_sent) == 4
+        payments_statuses_sent_payments_ids = [
+            payments_status_sent.paymentId for payments_status_sent in payments_statuses_sent
+        ]
+        for payment_to_be_marked_as_sent in payments_to_be_marked_as_sent:
+            assert payment_to_be_marked_as_sent.id in payments_statuses_sent_payments_ids
+
+        assert no_status_under_review_payment.id not in payments_statuses_sent_payments_ids
+        assert wrong_transaction_label_payment.id not in payments_statuses_sent_payments_ids
+
+    @patch("pcapi.scripts.payment.mark_payments_as_sent.logger")
+    def test_mark_payments_as_sent_does_not_fail_when_no_data(self, mocked_logger, app, db_session):
+        # Given
+        mark_payments_as_sent("transaction_label", 2)
+
+        # Then
+        mocked_logger.info.assert_called_with("No payments needed to be marked as sent")
```

Comment:
Je ne suis pas trop fan de tester le log. Une question de goût, peut-être. :) À la place, on pourrait tester qu'on n'a ajouté aucun paiement (parce que finalement, c'est surtout ça qu'on souhaite vérifier, le log n'est qu'un complément) :

```suggestion
         assert PaymentStatus.query.count() == 0
```